sdk/internal/go.mod

```diff
@@ -1,17 +1,20 @@
 module github.com/Azure/azure-sdk-for-go/sdk/internal
 
-go 1.18
+go 1.22.7
+
+toolchain go1.23.1
 
 require (
 	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.17.0
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.8.2
+	github.com/Azure/azure-sdk-for-go/sdk/tracing/azotel v0.4.0
```

Comment:
This will need to be removed as `azotel` requires a newer version of Go compared to the rest of our SDKs.

Reply:
@karenychen, this complicates our story about testing this within the azservicebus module then as well. Looks like we'll have to go with in-memory testing only for now.

Side note: @jhendrixMSFT, I'm going to start the work to move the stress tests out of the azserviecbus module because we will want to use azotel for those. The stress tests would never be published, so it should be okay to move those to the latest version of Go.

Reply:
Hm, I think we need to remove the otel propagation in these validators then because that's what's requiring the higher version. I will make the validator take in provider options so they can decide which propagator to use, or if they need to test propagation at all